pkg/metrics/utils.go

```diff
@@ -46,15 +48,17 @@ func mapConditionStatus(status gardenv1beta1.ConditionStatus) float64 {
 	}
 }
 
-func usedAsSeed(shoot *gardenv1beta1.Shoot) bool {
+func usedAsSeed(shoot *gardenv1beta1.Shoot, managedSeeds []*seedmanagementv1alpha1.ManagedSeed) bool {
 	if shoot.Namespace != constants.GardenNamespace {
 		return false
 	}
-	_, ok := shoot.Annotations[constants.AnnotationShootUseAsSeed]
-	if !ok {
-		return false
+	for _, ms := range managedSeeds {
```

Comment:
Idea for future optimization: instead of looping through the managed seeds for each shoot, create a map (namespace, shoot name) -> (true) and perform a hash based lookup when checking the existence of a managed seed resource for a given shoot.
Note that as there are not so many managed seeds objects, it is not obvious whether that would yield a significant performance benefit.

Reply:
I thought about that too since this is O(n^2) but like you said there are not many seeds so I don't think it should be a relevant performance impact. This is why I chose the easier implementation :)